File: src/lib/exercises/timed.ts
/**
 * Time-based movements record a HOLD in seconds, not reps — the deck + Nexus
 * label their "reps" field as "sec". Matched by name (planks, holds, carries).
 */
export function isTimedExercise(name: string | null | undefined): boolean {
  if (!name) return false
  return /\b(plank|hollow\s*hold|hold|dead\s*hang|wall\s*sit|l-?sit|carry)\b/i.test(name)
}

File: native/Packages/HelixCore/Sources/HelixCore/Exercises/Timed.swift
import Foundation

/// Time-based movements record a HOLD in seconds, not reps. Matched by name —
/// a port of `src/lib/exercises/timed.ts`, regex for regex.
public enum TimedExercise {
    private static let pattern = try! NSRegularExpression(
        pattern: #"\b(plank|hollow\s*hold|hold|dead\s*hang|wall\s*sit|l-?sit|carry)\b"#,
        options: .caseInsensitive
    )

    public static func isTimed(_ name: String?) -> Bool {
        guard let name, !name.isEmpty else { return false }
        return pattern.firstMatch(in: name, range: NSRange(location: 0, length: (name as NSString).length)) != nil
    }
}
